### ISSUE TITLE
Nodes can remain when a way is deleted

### ISSUE THREAD

**Issue by tomhughes** [tomhughes/trac-tickets#2785]
**[Submitted to the original trac issue database at 12.45pm, Tuesday, 9th March 2010]**

Version: svn 20382
OS: Linux

Create two ways.  The second way should have it's start and end points attached to the first way e.g.

```

   /\
  /  \
 /----\
/      \


/\   represents way 1
---  represents way 2
```

Delete way 2, then delete way 1.  You'll be left with the nodes that were created when way 2 was originally joined to way 1.

**Comment by tomhughes** [comment]
**Author: koying**
**[Added to the original trac issue at 1.41pm, Tuesday, 9th March 2010]**

(In [20386]) FIX : Do not count deleted parents when removing children nodes (fixes #2785)
FIX : Make sure not to write to the log after it has been closed (fixes #2783)

**Comment by tomhughes** [comment]
**Author: koying**
**[Added to the original trac issue at 1.45pm, Tuesday, 9th March 2010]**

(In [20387]) FIX : Do not count deleted parents when removing children nodes (fixes #2785)
FIX : Make sure not to write to the log after it has been closed (fixes #2783)

**Comment by tomhughes** [comment]
**Author: nick@sqrt.co.uk**
**[Added to the original trac issue at 5.25pm, Tuesday, 9th March 2010]**

Whilst the fix did have the desired outcome, it's created a new issue:

Create the two ways as originally described.  Now delete way 1 and you'll find that way 2 disappears at the same time.

**Comment by tomhughes** [comment]
**Author: koying**
**[Added to the original trac issue at 9.35am, Wednesday, 10th March 2010]**

(In [20406]) FIX : think, man, think... (closes #2785)

**Comment by tomhughes** [comment]
**Author: koying**
**[Added to the original trac issue at 9.36am, Wednesday, 10th March 2010]**

(In [20407]) FIX : think, man, think... (closes #2785)
